File: Assets/ProjectGame/Lib/data/class_data_id/RoleType/RoleTypeRow.js
import { BaseClassDataRow } from '../BaseClassDataRow.js';
import { Faction } from '../../enum/Faction/Faction.js';

export class RoleTypeRow extends BaseClassDataRow {
    constructor() {
        super();
        this.name = ""; // 役職名
        this.faction_id = Faction.NONE; // 所属ID
        this.use = false; // 使用フラグ
    }

    read(reader) {
        this.name = reader.readString();
        this.faction_id = Faction.fromInt(reader.readInt32());
        this.use = reader.readBoolean();
    }

    static fromJson(data) {
        const self = new this();
        self.name = data.name ?? "";
        self.faction_id = data.faction_id ?? Faction.NONE;
        self.use = data.use ?? false;
        return self;
    }
}

File: Assets/ProjectGame/Lib/data/class_data_id/RoleType/RoleTypeRow.py
from ..BaseClassDataRow import BaseClassDataRow
from ...enum.Faction.Faction import Faction

class RoleTypeRow(BaseClassDataRow):
    def __init__(self):
        super().__init__()
        self.name = ""  # 役職名
        self.faction_id = Faction.NONE  # 所属ID
        self.use = False  # 使用フラグ

    def read(self, reader):
        self.name = reader.read_string()
        self.faction_id = Faction(reader.read_int32())
        self.use = reader.read_bool()

    @classmethod
    def from_json(cls, data: dict):
        self = cls()
        self.name = data.get('name', "")
        self.faction_id = data.get('faction_id', Faction.NONE)
        self.use = data.get('use', False)
        return self
